src/com/palmergames/bukkit/towny/event/actions/TownyExplosionDamagesEntityEvent.java

```diff
@@ -0,0 +1,91 @@
+package com.palmergames.bukkit.towny.event.actions;
+
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
+
+/**
+ * Part of the API which lets Towny's war and other plugins modify Towny's
+ * plot-permission-decision outcomes.
+ * 
+ * TownyExplosionDamagesEntityEvents are thrown when an explosion 
+ * occurs in a Towny world, causing damage to an entity.
+ * 
+ * <br> - When an entity explosion, block explosion or lightning damages an entity.
+ * <br> - When an explosion would damage a Hanging entity.
+ * <br> - When a pig is zapped by lightning.
+ * <br> - When an explosion would damage a Vehicle.
+ * 
+ * @param location - Location of the entity being damaged.
+ * @param entity - Entity getting exploded.
+ * @param cause - DamageCause.
+ * @param canExplode - Whether Towny will cancel this already.
+ * 
+ * @author LlmDl
+ */
+public class TownyExplosionDamagesEntityEvent extends Event implements Cancellable {
+
+	private static final HandlerList handlers = new HandlerList();
+	private final Location location;
+	private final Entity entity;
+	private final DamageCause cause;
+	private boolean canExplode;
+
+	/**
+	 * Event thrown when an explosion damages an entity. 
+	 * Use ignorecancelled = true in order to filter out explosions Towny will already have stopped.
+	 * 
+	 * @param location - Location of the entity being damaged.
+	 * @param harmedEntity - Entity getting exploded.
+	 * @param cause - DamageCause.
+	 * @param canExplode - Whether Towny will cancel this already.

```

Comment:
I feel like `canExplode` should be renamed because it's actual value dictates whether the event is cancelled, so it's really more can't explode than can. Either that or the value should be inverted so canExplode returns false if the event should be cancelled.